### ISSUE TITLE
[java] UnnecessaryImport: False positive when Javadoc {@link} references an array type

### ISSUE THREAD

**Issue by Emilyaxe** [pmd/pmd#6651]
**Affects PMD Version:** 7.24.0

**Rule:** UnnecessaryImport — https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#unnecessaryimport

**Description:**

`UnnecessaryImport` incorrectly marks an import as unused when it is only referenced in Javadoc `{@link}` using an array type (e.g. `Type[]`), while correctly recognizing the same import as used when referenced as `Type`. This leads to inconsistent false positives.

**Code Sample demonstrating the issue:**

```java
// BEFORE — no violation reported (correct)
package demo;

import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.attribute.FileAttribute;

/**
 * See {@link Files#createDirectory(Path, FileAttribute)} for directory creation.
 */
public class Test {}
```

```java
// AFTER — UnnecessaryImport reported on line 5 (false-positive)
package demo;

import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.attribute.FileAttribute;

/**
 * See {@link Files#createDirectory(Path, FileAttribute[])} for directory creation.
 */
public class Test {}
```

**Expected outcome:** No `UnnecessaryImport` violation on the AFTER snippet — an import referenced from a Javadoc `{@link}` tag should be considered used regardless of whether the reference is written as `T` or `T[]`.

**Actual outcome:** PMD reports `Unused import 'java.nio.file.attribute.FileAttribute'` at line 5 of AFTER, while correctly reporting nothing on BEFORE.

**Running PMD through:** CLI

**Comment by UncleOwen** [comment]
Related to https://github.com/pmd/pmd/issues/4325
